src/main/java/emissary/util/TimeUtil.java

```diff
@@ -250,24 +253,30 @@ public static ZonedDateTime getZonedDateFromISO8601(@Nullable final String dateS
      *
      */
     public static String convertHexDate(String hexDate) {
-        String hexNumber = hexDate.substring(HEX_PREFIX_LEN);
-        String dateHex = hexNumber.substring(0, 8);
-        String timeHex = hexNumber.substring(8, 16);
+        Matcher m = P.matcher(hexDate);
 
-        long daysToAdd = Long.parseLong(dateHex, HEX_RADIX);
-        long millisToAdd = Math.round(Long.parseLong(timeHex, HEX_RADIX) * 10 / 3.0);
+        if (m.find()) {
+            String dateHex = m.group(1);
+            String timeHex = m.group(2);
+            long daysToAdd = Long.parseLong(dateHex, HEX_RADIX);
+            long millisToAdd = Math.round(Long.parseLong(timeHex, HEX_RADIX) * 10 / 3.0);
```

Comment:
Let's a new inline comment here to explain the non-obvious math

```suggestion
            // timeHex represents the number of ticks (1/300 of a second) since midnight
            long millisToAdd = Math.round(Long.parseLong(timeHex, HEX_RADIX) * 10 / 3.0);
```